### PR TITLE
fix(graph-gateway): construct the network snapshot from a single subgraphs paginated query

### PR DESCRIPTION
**Pull request by LNSD**
The number of entities that can be returned by a graphql query is limited. In some cases, the number of allocations associated with an indexer can be greater than that limit, so the initial approach I took is not viable. 

This has motivated a change in the network module subgraph client and all the associated data processing pipelines.

### CODE REVIEW

File: graph-gateway/src/network/internal.rs [hunk 1]
```diff
@@ -40,47 +41,25 @@ pub async fn fetch_update(
...
+        NETWORK_TOPOLOGY_FETCH_TIMEOUT,
+        fetch_and_pre_process_subgraph_info(client),
+    )
+    .await
+    {
+        // If the fetch timed out, return an error
+        Err(_) => Err(anyhow!("info fetch timed out")),
+        Ok(resp) => match resp {
+            // If the fetch failed, return an error
+            Err(err) => Err(anyhow!(" info fetch failed: {err}")),
```
Comment (Theodus) [inline graph-gateway/src/network/internal.rs:54]:
```suggestion
            Err(err) => Err(anyhow!("info fetch failed: {err}")),
```

File: graph-gateway/src/network/subgraph_client.rs [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        let query = format!(
+            r#"
+            subgraphs(
+                block: $block
+                orderBy: id, orderDirection: asc
+                first: $first
+                where: {{
+                    id_gt: $last
+                    entityVersion: 2
+                    versionCount_gt: 0
```
Comment (Theodus) [inline graph-gateway/src/network/subgraph_client.rs:125]: Shouldn't this be `versionCount_gt: 1` to meet the requirements above?
> Get all subgraphs with at least one version.

File: graph-gateway/src/network/subgraph_client.rs [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+                versions(orderBy: version, orderDirection: desc) {{
+                    version
+                    subgraphDeployment {{
+                        ipfsHash
+                        manifest {{
+                            network
+                            startBlock
+                        }}
+                        indexerAllocations(
+                            first: 250
```
Comment (Theodus) [inline graph-gateway/src/network/subgraph_client.rs:140]: why 250? Seems like a lot

**Review by Theodus — APPROVED** [review]
LGTM

File: graph-gateway/src/network/subgraph_client.rs (hunk 2 above)
Comment (LNSD) [reply graph-gateway/src/network/subgraph_client.rs:125]: This means `count > 0`, which would be equivalent to `count >= 1`.

Fro clarity, I can change it with `versionCount_gte: 1` (count >= 1).

File: graph-gateway/src/network/subgraph_client.rs (hunk 3 above)
Comment (LNSD) [reply graph-gateway/src/network/subgraph_client.rs:140]: I did a test and forgot to restore the value to the original `100`. 

If I remember correctly, the largest number of indexer allocations was around ~60, which is within the original limit of 100.

Comment (LNSD) [reply graph-gateway/src/network/subgraph_client.rs]: Addressed in [15d3252](https://github.com/edgeandnode/gateway/pull/785/commits/15d3252ecedab95bf24ad9d820d0eac648d18305)

File: graph-gateway/src/network/subgraph_client.rs (hunk 2 above)
Comment (LNSD) [reply graph-gateway/src/network/subgraph_client.rs]: Addressed in [15d3252](https://github.com/edgeandnode/gateway/pull/785/commits/15d3252ecedab95bf24ad9d820d0eac648d18305)

File: graph-gateway/src/network/internal.rs (hunk 1 above)
Comment (LNSD) [reply graph-gateway/src/network/internal.rs]: Addressed in [15d3252](https://github.com/edgeandnode/gateway/pull/785/commits/15d3252ecedab95bf24ad9d820d0eac648d18305)

**Comment by LNSD** [comment]
Part of https://github.com/edgeandnode/gateway/issues/791

**Pull request merged**